lib/appmap/value_inspector.rb

```diff
@@ -32,8 +34,12 @@ def detect_schema(value, max_depth: MAX_DEPTH, type_info: {}, observed_values: S
           detect_schema(next_value, **{ max_depth: max_depth, type_info: value_schema, observed_values: observed_values, depth: depth + 1 })
         end.compact
         type_info[:properties] = properties unless properties.empty?
-      elsif value.respond_to?(:first)
-        detect_schema(value.first, **{ max_depth: max_depth, type_info: type_info, observed_values: observed_values, depth: depth + 1 })
+      elsif value.respond_to?(:map)
+        type_info[:properties] = value.map do |next_value|
+          value_schema = { class: best_class_name(next_value) }
+          detect_schema(next_value, **{ max_depth: max_depth, type_info: value_schema, observed_values: observed_values, depth: depth + 1 })
+        end
+        type_info[:properties] = type_info[:properties].compact.uniq { |schema| schema[:class] }
```

Comment:
I am not sure that this behavior of emitting an array of `properties` object for every array entry is really compliant with the appmap specification (at least, with the intent of it) or with the behavior of the OpenAPI generation tool. It might be better to do a merge of the `schema` here and avoid setting properties to an array. It's probably causing problems downstream because the OpenAPI generation code is not anticipating that this will happen?

Reply:
I've switched to something a little more in line with OpenAPI - adding an `items` property which describes the type(s) held by the array.